.github/workflows/check_files.py

```diff
@@ -0,0 +1,159 @@
+    """
+    Check for unauthorized file changes.
+    Args:
+        changed_files (list): List of changed files.
+        sensitive_files (list): List of sensitive files.
+    Returns:
+        list: Unauthorized changes.
+
+    """
+
+    unauthorized_changes = [file for file in changed_files if any(glob.fnmatch.fnmatch(file, sf) for sf in sensitive_files if sf)]
+    return ScriptResult(len(unauthorized_changes), unauthorized_changes)
+    
+
+def _arg_parser_resolver():
+    """Resolve the CLI arguments provided by the user.
+    Args:
+        None
+    Returns:
+        result: Parsed argument object
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--base_branch",
+        type=str,
+        required=True,
+        help="Base branch where a pull request should be made."
+    )
+    parser.add_argument(
+        "--pr_branch",
+        type=str,
+        required=True,
+        help="PR branch from where the pull request is made.",
+    )
+    parser.add_argument(
+        "--file_count",
+        type=int,
+        default=20,
+        help="Number of files changes allowed in a single commit"
+    )
+    parser.add_argument(
+        "--sensitive_files",
+        type=str,
+        required=True,
+        help="Path to a file containing a list of sensitive files and directories."
+    )
+    return parser.parse_args()
+
+def main():
+    """
+    Execute the script's main functionality.
+    This function serves as the entry point for the script. It performs
+    the following tasks:
+    1. Validates and retrieves the base branch and PR commit from
+       command line arguments.
+    2. Counts the number of changed files between the specified branches.
+    3. Checks if the count of changed files exceeds the acceptable
+       limit (20).
+    4. Checks for unauthorized file changes based on a list of sensitive files.
+    5. Provides informative messages based on the analysis.
+    Raises:
+        SystemExit: If an error occurs during execution.
+    """
+
+    args = _arg_parser_resolver()
+
+    base_branch = args.base_branch
+    pr_branch = args.pr_branch
+
+    print(f"You are trying to merge on branch: {base_branch}")
+    print(f"You are making a commit from your branch: {pr_branch}")
+
+    # Read sensitive files from the provided file
+    with open(args.sensitive_files, 'r') as sensitive_file:
+        sensitive_files = [line.strip() for line in sensitive_file]
+
+    # Count changed files and check for unauthorized changes
+    result = _count_changed_files(base_branch, pr_branch)
+    print(f"Number of changed files: {result.file_count}")
+
+    unauthorized_changes = _check_unauthorized_changes(result.unauthorized_changes, sensitive_files)
+    print(f"Unauthorized changes: {unauthorized_changes}")
+
+    # Check if the count exceeds the allowed limit
+    if result.file_count > args.file_count:
+        print("Error: Too many files (greater than 20) changed in the pull request.")
```

Comment:
1. `20` should be replaced with the argparse variable
1. List the number of offending files here too as part of the error message